pkgs/shells/zsh/default.nix

```diff
@@ -55,7 +55,9 @@ stdenv.mkDerivation {
   checkFlags = map (T: "TESTNUM=${T}") (lib.stringToCharacters "ABCDEVW");
 
   # XXX: think/discuss about this, also with respect to nixos vs nix-on-X
-  postInstall = ''
+  postInstall = lib.optional stdenv.isDarwin ''
+    make install.bin install.modules install.fns
+    '' ++ lib.optional stdenv.isLinux ''
```

Comment:
This is removing too much stuff, I think?

You just want to apply the switch to the `make ...` line, not the entire `postInstall`

Reply:
Sorry for missing that! Running nixpkgs-review now.